src/LibVLCSharp/Shared/MediaPlayer.cs

```diff
@@ -635,10 +633,19 @@ public MediaPlayer(Media media)
         {
             get
             {
-                var mediaPtr = Native.LibVLCMediaPlayerGetMedia(NativeReference);
-                return mediaPtr == IntPtr.Zero ? null : new Media(mediaPtr);
+                return _media;
+            }
+            set
+            {
+                if(_media?.NativeReference != IntPtr.Zero)
```

Comment:
Either media is null or it's not, but if it's not null, I don't think its native reference can be Zero. Either way, it's up to _media.Dispose() to check that (same in MediaPlayer.Dispose)